tests/e2e/framework/kubernetes.go

```diff
@@ -135,6 +137,52 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
 	}, nil
 }
 
+func (k *KubeInfo) GetGateway() string {
+	return k.Ingress
+}
+
+func (k *KubeInfo) AddApp(appYaml string) {
+	demo := App{AppYaml: util.GetResourcePath(filepath.Join(rulesDir, appYaml)), KubeInject: true}
+	k.AppManager.AddApp(&demo)
+}
+
+func (k *KubeInfo) CheckRunning() bool {
+	return util.CheckPodsRunning(k.Namespace)
+}
+
+func (k *KubeInfo) DeleteRule(rule string) error {
```

Comment:
nit: Might be helpful to clarify what the parameter is: `ruleYaml`? `ruleName`?  Same for the other methods below.